src/librustc/session/mod.rs

```diff
@@ -336,6 +336,10 @@ impl NodeIdAssigner for Session {
     fn peek_node_id(&self) -> NodeId {
         self.next_node_id.get().checked_add(1).unwrap()
     }
+
+    fn diagnostic(&self) -> &errors::Handler {
+        self.diagnostic()
```

Comment:
> There's no way to say "I want an inherent impl and only an inherent impl"

That seems good. Then you can factor out an inherent method into a trait method without a breaking change, is that true?